src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php

```diff
@@ -23,15 +24,24 @@ class PreventRequestsDuringMaintenance
      */
     protected $except = [];
 
+    /**
+     * The maintenance mode implementation. Can be used to check
+     * if the application is under maintenance.
+     *
+     * @var \Illuminate\Foundation\MaintenanceMode
+     */
+    protected $maintenanceMode;
+
     /**
      * Create a new middleware instance.
      *
      * @param  \Illuminate\Contracts\Foundation\Application  $app
-     * @return void
+     * @param  \Illuminate\Foundation\MaintenanceMode  $maintenanceMode
```

Comment:
```suggestion
     * @param  \Illuminate\Foundation\MaintenanceMode  $maintenanceMode
     * @return void
```

Reply:
I wonder if we can add a `maintenance` method to the `Application` class that returns the `Maintenance` instance? That way we don't need an extra dependency here. And you can call `$this->app->maintenance()->` for these calls.

Reply:
That would indeed remove the dependency and make it more available within the application as a whole. I will add it to the Application class